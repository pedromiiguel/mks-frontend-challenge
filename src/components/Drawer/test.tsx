import { customRender } from '@/utils/tests/customRender';
import { fireEvent, screen } from '@testing-library/react';

import { Drawer } from '.';

describe('<Drawer />', () => {
  it('should render the drawer', () => {
    const onCloseDrawerMock = jest.fn();

    const { container } = customRender(
      <Drawer isOpen={true} onCloseDrawer={onCloseDrawerMock} />
    );

    expect(screen.getByRole('heading', { name: 'Carrinho de compras' }));
    expect(screen.getByRole('button', { name: 'Fechar carrinho de compras' }));
    expect(screen.getByText('Total:'));

    expect(container.firstChild).toMatchSnapshot();
  });

  it('calls onCloseDrawer when close button is clicked', () => {
    const onCloseDrawerMock = jest.fn();
    customRender(<Drawer isOpen={true} onCloseDrawer={onCloseDrawerMock} />);

    fireEvent.click(
      screen.getByRole('button', { name: /Fechar carrinho de compras/i })
    );

    expect(onCloseDrawerMock).toHaveBeenCalledTimes(1);
  });

  it('calls onCloseDrawer when overlay is clicked', () => {
    const onCloseDrawerMock = jest.fn();

    const { debug } = customRender(
      <Drawer isOpen={true} onCloseDrawer={onCloseDrawerMock} />
    );

    const overlay = screen.getByLabelText(
      'Sobreposição do carrinho de compras'
    );

    fireEvent.click(overlay);

    expect(onCloseDrawerMock).toHaveBeenCalledTimes(1);
  });

  it('should drawer is open disabled body overflow', () => {
    const onCloseDrawerMock = jest.fn();

    customRender(<Drawer isOpen={true} onCloseDrawer={onCloseDrawerMock} />);

    expect(document.body).toHaveStyle({ overflow: 'hidden' });
  });
});
