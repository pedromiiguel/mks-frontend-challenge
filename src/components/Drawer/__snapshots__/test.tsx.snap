// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer /> should render the drawer 1`] = `
.c0 {
  z-index: 40;
  position: fixed;
  background-color: #0F52BA;
  top: 0;
  right: 0;
  bottom: 0;
  max-width: 48rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 90%;
  max-height: 100vh;
  box-shadow: -5px 0px 6px rgba(0,0,0,0.13);
  -webkit-animation-name: open;
  animation-name: open;
  -webkit-animation-duration: 0.5s;
  animation-duration: 0.5s;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.c3 {
  background-color: #000000;
  width: 3.8rem;
  height: 3.8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
  border: 0;
  cursor: pointer;
}

.c3 > svg {
  width: 1.5rem;
  height: 2.0rem;
  fill: #FFFFFF;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 3.2rem;
}

.c2 {
  font-weight: 700;
  font-size: 3.2rem;
  line-height: 3.3rem;
  width: 100%;
  max-width: 19.6rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  gap: 2.4rem;
  margin-bottom: 2.4rem;
}

.c4 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow-y: auto;
  padding: 3.2rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 3.2rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c8 {
  font-weight: 700;
  font-size: 3.2rem;
  line-height: 1.5rem;
}

.c9 {
  font-weight: 700;
  font-size: 3.2rem;
  line-height: 1.5rem;
}

.c10 {
  background-color: #000000;
  font-size: 2.0rem;
  font-weight: 700;
  padding: 2.4rem;
  margin-top: 3.2rem;
}

@media (max-width:450px) {
  .c1 {
    padding: 2.4rem;
  }
}

@media (max-width:450px) {
  .c4 {
    padding: 0 2.4rem;
  }
}

@media (max-width:450px) {
  .c6 {
    padding: 2.4rem;
  }
}

@media (max-width:450px) {
  .c8 {
    font-size: 2.0rem;
  }
}

@media (max-width:450px) {
  .c9 {
    font-size: 2.0rem;
  }
}

@media (max-width:450px) {
  .c10 {
    margin-top: 2.4rem;
  }
}

<div
  aria-hidden="false"
  aria-label="Carrinho de compras"
  class="c0"
>
  <header
    class="c1"
  >
    <h2
      class="c2"
    >
      Carrinho de compras
    </h2>
    <button
      aria-label="Fechar carrinho de compras"
      class="c3"
    >
      <svg
        aria-label="Fechar"
        fill="none"
        height="8"
        role="img"
        viewBox="0 0 8 8"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0.266 8L3.374 3.968L3.36 4.458L0.406 0.649999H1.526L3.948 3.8L3.528 3.786L5.95 0.649999H7.042L4.06 4.5L4.074 3.968L7.21 8H6.076L3.514 4.64L3.906 4.696L1.386 8H0.266Z"
        />
      </svg>
    </button>
  </header>
  <div
    class="c4"
  >
    <div
      class="c5"
    />
  </div>
  <footer
    class="c6"
  >
    <div
      class="c7"
    >
      <h4
        class="c8"
      >
        Total: 
      </h4>
      <div
        class="0 c9"
      >
        R$Â 0
      </div>
    </div>
  </footer>
  <button
    aria-label="Finalizar compra"
    class="1 c10"
  >
    Finalizar Compra
  </button>
</div>
`;
